src/features/events/components/EventsList/index.tsx

```diff
@@ -12,15 +12,23 @@ import { useEvents } from '../../hooks/useEvents'
 import type { Event } from '../../types'
 
 const sorts = {
-  ascending: (a: Event, b: Event) => (a.startsAt < b.startsAt ? -1 : 1),
-  descending: (a: Event, b: Event) => (a.startsAt > b.startsAt ? -1 : 1),
+  asc: (a: Event, b: Event) => (a.startsAt < b.startsAt ? -1 : 1),
+  desc: (a: Event, b: Event) => (a.startsAt > b.startsAt ? -1 : 1),
 }
 
 const filters = {
   future: (event: Event) => isAfter(new Date(event.startsAt), new Date()),
   past: (event: Event) => isBefore(new Date(event.startsAt), new Date()),
 }
 
+const { ALL, FUTURE, PAST } = FilterType
+
+const listBuilders = {
+  [ALL]: (events: Event[]) => [...events].sort(sorts.asc),
+  [FUTURE]: (events: Event[]) => events.filter(filters.future).sort(sorts.asc),
+  [PAST]: (events: Event[]) => events.filter(filters.past).sort(sorts.desc),
+}
```

Comment:
Okay, what is a "List Builder"? Well, we have these three "flavors" of lists (All, Future, Past), and each is a different combination of filter + sort:

- **All:** no filter, sort ascending
- **Future:** filter in those in the future, sort ascending
- **Past:** filter in those in the past, sort descending

The list builders is just an easier way for us to _pick_ these functions based on a `FilterType`. It's a map.

Reply:
this is really super clean! 👍